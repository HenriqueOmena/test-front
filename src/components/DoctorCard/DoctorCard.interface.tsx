export interface IAddress {
  line1: string;
  line2: string;
  postalCode: string;
  city: string;
  country: string;
}

export interface IDoctorCard {
  id: string;
  name: string;
  address: IAddress;
}
