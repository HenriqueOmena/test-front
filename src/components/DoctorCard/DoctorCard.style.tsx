import { makeStyles } from "@material-ui/core";

const DoctorCardStyle = makeStyles(() => ({
  card: {
    width: "100%",
    maxWidth: 750,
    borderBottom: "solid 1px lightgray"
  },
  header: {
    display: "flex",
    alignItems: "center",
    justifyContent: "space-around",
    padding: 20,
    paddingBottom: 0
  },
  buttonAvailabilities: {
    padding: 10,
    margin: 10,
    marginLeft: 20,
    marginRight: 0,
    minWidth: 160
  },
  collapse: {
    padding: 20,
    paddingTop: 0
  },
  availability: {
    marginTop: 10
  },
  availabilityItem: {
    display: "flex",
    justifyContent: "space-between",
    alignItems: "center",
    marginTop: 5
  }
}));

export default DoctorCardStyle;
