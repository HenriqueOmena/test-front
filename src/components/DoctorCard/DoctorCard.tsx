import React, { useState } from "react";
import { Button, Collapse } from "@material-ui/core";
import DoctorCardStyle from "./DoctorCard.style";
import { IDoctorCard } from "./DoctorCard.interface";
import { useRequest } from "hooks/useRequest";
import { AddressDetails } from "components/AddressDetails";
import { DoctorAvatar } from "components/DoctorAvatar";
import { DoctorAvailability } from "components/DoctorAvailability";

const DoctorCard: React.FC<IDoctorCard> = (props) => {
  const { id, address, name } = props;
  const classes = DoctorCardStyle();
  const [open, setOpen] = useState(false);
  const { apiData, isLoading, execute } = useRequest("GET", `availabilities?doctorId=${id}`, false);

  const handleClick = () => {
    setOpen(!open);
    if (apiData.length === 0) execute();
  };

  return (
    <div className={classes.card}>
      <div className={classes.header}>
        <DoctorAvatar name={name} />
        <AddressDetails {...address} />
        <Button
          variant="contained"
          color="secondary"
          onClick={handleClick}
          className={classes.buttonAvailabilities}
        >
          Availabilities
        </Button>
      </div>

      <div className={classes.collapse}>
        {isLoading && <span>Loading.....</span>}

        <Collapse in={open} timeout="auto" unmountOnExit>
          <DoctorAvailability data={apiData} {...props} />
        </Collapse>
      </div>
    </div>
  );
};

export default DoctorCard;
