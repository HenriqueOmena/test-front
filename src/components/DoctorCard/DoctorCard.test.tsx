import { fireEvent, render } from "@testing-library/react";
import DoctorCard from "./DoctorCard";

describe("Component - DoctorCard", () => {
  const propMock = {
    id: "someId",
    name: "Henrique Omena",
    address: {
      city: "Funchal",
      country: "Portugal",
      line1: "Santa",
      line2: "Rita",
      postalCode: "0099"
    }
  };

  let component = <DoctorCard {...propMock} />;

  it("Should Match SnapShot", () => {
    expect(component).toMatchSnapshot();
  });

  it("Should render with prop name", async () => {
    const { getByText } = render(component);
    const nameText = getByText(propMock.name);

    expect(nameText).toBeVisible();
  });

  it("Should Click on Availabilities Button", async () => {
    const { getByText } = render(component);
    const button = getByText("Availabilities");
    fireEvent.click(button);
  });
});
