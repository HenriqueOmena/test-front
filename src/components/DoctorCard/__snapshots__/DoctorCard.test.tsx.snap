// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - DoctorCard Should Match SnapShot 1`] = `
<DoctorCard
  address={
    Object {
      "city": "Funchal",
      "country": "Portugal",
      "line1": "Santa",
      "line2": "Rita",
      "postalCode": "0099",
    }
  }
  id="someId"
  name="Henrique Omena"
/>
`;
